hail/src/main/scala/is/hail/expr/ir/BlockMatrixIR.scala

```diff
@@ -270,6 +270,17 @@ case class BlockMatrixMap(child: BlockMatrixIR, eltName: String, f: IR, needsDen
   override protected[ir] def execute(ctx: ExecuteContext): BlockMatrix = {
     val prev = child.execute(ctx)
 
+    val functionArgs = f match {
+      case ApplyUnaryPrimOp(_, arg1) => IndexedSeq(arg1)
+      case Apply(_, _, args, _) => args
+      case ApplyBinaryPrimOp(_, l, r) => IndexedSeq(l, r)
+    }
+
+    assert(functionArgs.forall(ir => IsConstant(ir) || ir.isInstanceOf[Ref]),
```

Comment:
can this replace the match error below?

Reply:
The reason this is necessary is because if you do something like `bm.map(lambda x: hl.abs(x + 1))` with the match, it will just apply the outermost function (the `abs`), not transitively apply the series of applies. But the other match is necessary because we don't support every possible apply function on BlockMatrix, just ones we've special cased.